server/runner/taskrun/scheduler.go

```diff
@@ -170,12 +170,22 @@ func (s *Scheduler) Run(ctx context.Context, wg *sync.WaitGroup) {
 						continue
 					}
 
+					if s.stateCfg.InstanceOutstandingConnections[task.Database.InstanceID] >= state.InstanceMaximumConnectionNumber {
```

Comment:
read should also be guarded by the lock

Reply:
Done. Yup, was planning to use inaccurate number to avoid unpleasant unlocking code.